.github/workflows/build-and-test.yml

```diff
@@ -69,15 +69,15 @@ jobs:
         if: ${{ matrix.os != 'windows-latest' }}
         uses: 'swift-actions/setup-swift@v1'
         with:
-          version: 5.9
+          version: 5.10
 
       - name: Setup Swift (Windows)
         if: ${{ matrix.os == 'windows-latest' }}
         uses: 'compnerd/gha-setup-swift@main'
         with:
           github-repo: thebrowsercompany/swift-build
           release-asset-name: installer-amd64.exe
-          release-tag-name: 20231010.3
```

Comment:
I think we should be using [this pattern](https://github.com/hylo-lang/hylo/blob/08f626d0c66c83c9eec33a9880736547385e26b7/.github/workflows/build-and-test.yml#L231-L239).